src/spi.rs

```diff
@@ -1923,16 +2112,32 @@ extern "C" fn spi_notify(transaction: *mut spi_transaction_t) {
     }
 }
 
-fn copy_operation<'b>(operation: &'b mut Operation<'_, u8>) -> Operation<'b, u8> {
+fn copy_operation<'b>(operation: &'b mut Operation<'_>) -> Operation<'b> {
     match operation {
         Operation::Read(read) => Operation::Read(read),
+        Operation::ReadWithWidth(read, line_width) => Operation::ReadWithWidth(read, *line_width),
         Operation::Write(write) => Operation::Write(write),
+        Operation::WriteWithWidth(write, line_width) => {
+            Operation::WriteWithWidth(write, *line_width)
+        }
         Operation::Transfer(read, write) => Operation::Transfer(read, write),
         Operation::TransferInPlace(write) => Operation::TransferInPlace(write),
         Operation::DelayNs(delay) => Operation::DelayNs(*delay),
     }
 }
 
+fn copy_ehal_operation<'b>(
```

Comment:
Nit: You can also just implement `impl<'a> From<&embedded_hal::spi::Operation<'a, u8>> for Operation<'a>`

Nit 2: I don't think you need `operation: &'b mut <Operation<'_>` to be mutable.

Reply:
Same as below, it still has to be `&mut Operation`. And I suppose it could be `From`, but I wanted to keep minimal changes here